executor/builder.go

```diff
@@ -571,12 +571,23 @@ func (b *executorBuilder) buildSelection(v *plan.Selection) Executor {
 	default:
 		src = b.build(x)
 	}
-
 	if len(v.Conditions) == 0 {
 		v.Conditions = oldConditions
+		switch x := src.(type) {
```

Comment:
1. Why reorder this during executor building phase?
2. Are you sure the c.Offset and s.Postition have the same meanings ?
3. What if that is a correlated column ?